batch7crud-roll8/src/main/java/com/lftechnology/batch7crud/controller/EmployeesController.java

```diff
@@ -0,0 +1,210 @@
+    if (path == null) {
+      fetch(request, response);
+    }
+    else {
+      String[] parts = path.split("/");
+      if (parts.length == 0) {
+        fetch(request, response);
+      }
+      else if (parts.length == 2 && "create".equals(parts[1])) {
+        create(request, response);
+      }
+      else if (parts.length == 2) {
+        view(request, response);
+      }
+      else if (parts.length == 3 && "edit".equals(parts[2])) {
+        edit(request, response);
+      }
+      else {
+        pageNotFound(request, response);
+      }
+    }
+  }
+
+  @Override
+  protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+    String path = request.getPathInfo();
+    if (path == null) {
+      request.getServletContext().getRequestDispatcher("/WEB-INF/views/employees/index.jsp").forward(request, response);
+    }
+    else {
+      String[] parts = path.split("/");
+      if ("create".equals(parts[1])) {
+        createProcess(request, response);
+      }
+      else if ("edit".equals(parts[2])) {
+        int id = Integer.parseInt(parts[1]);
+        editProcess(request, response, id);
+      }
+      else if ("delete".equals(parts[2])) {
+        int id = Integer.parseInt(parts[1]);
+        deleteProcess(request, response, id);
+      }
+    }
+  }
+
+  private void create(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+    request.getServletContext().getRequestDispatcher("/WEB-INF/views/employees/new.jsp").forward(request, response);
+  }
+
+  private void edit(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+    try {
+      int id = parameterValueAsInt(request, 2);
+      EmployeeService employeeService = new EmployeeService();
+      Employee employee = employeeService.fetchById(id);
+      request.setAttribute("employee", employee);
+      RequestDispatcher dispatcher = request.getRequestDispatcher("/WEB-INF/views/employees/edit.jsp");
+      dispatcher.forward(request, response);
+    }
+    catch (DataException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      errorPage(request, response, e.getMessage());
+    }
+
+  }
+
+  private void createProcess(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+    try {
+      String name = request.getParameter("name");
+      String address = request.getParameter("address");
+      String designation = request.getParameter("designation");
+      String phone = request.getParameter("phone");
+      EmployeeService employeeService = new EmployeeService();
+
+      Employee employee = new Employee();
+      employee.setName(name);
+      employee.setAddress(address);
+      employee.setDesignation(designation);
+      employee.setPhone(phone);
+
+      employeeService.create(employee);
+      response.sendRedirect("/employees");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "/employees" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)